cfn-resources/search-deployment/mongodb-atlas-searchdeployment.json

```diff
@@ -0,0 +1,94 @@
+{
+  "additionalProperties": false,
+  "description": "Returns, adds, edits, and removes Atlas Search indexes for the specified cluster. Also returns and updates user-defined analyzers for the specified cluster.",
+  "handlers": {
+    "create": {
+      "permissions": [
+        "secretsmanager:GetSecretValue"
+      ]
+    },
+    "read": {
+      "permissions": [
+        "secretsmanager:GetSecretValue"
+      ]
+    },
+    "update": {
+      "permissions": [
+        "secretsmanager:GetSecretValue"
+      ]
+    },
+    "delete": {
+      "permissions": [
+        "secretsmanager:GetSecretValue"
+      ]
+    }
+  },
+  "definitions": {
+    "ApiSearchDeploymentSpec": {
+      "type": "object",
+      "properties": {
+        "InstanceSize": {
+          "type": "string"
+        },
+        "NodeCount": {
+          "type": "integer"
+        }
+      },
+      "required": [
+        "InstanceSize",
+        "NodeCount"
+      ],
+      "additionalProperties": false
+    }
+  },
+  "properties": {
+    "Profile": {
+      "type": "string",
+      "default": "default"
+    },
+    "ClusterName": {
+      "type": "string"
+    },
+    "ProjectId": {
+      "type": "string"
+    },
+    "Id": {
+      "type": "string"
+    },
+    "Specs": {
+      "insertionOrder": false,
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ApiSearchDeploymentSpec",
+        "type": "object"
+      }
+    },
+    "StateName": {
+      "type": "string"
+    }
+  },
+  "primaryIdentifier": [
+    "/properties/ProjectId",
+    "/properties/ClusterName",
+    "/properties/Id"
+  ],
```

Comment:
Have not added Profile within the primary identifier as I am not finding a reason why we actually add it for all our resources.